clients/drcachesim/scheduler/scheduler.h

```diff
@@ -299,41 +346,52 @@ template <typename RecordType, typename ReaderType> class scheduler_tmpl_t {
         {
         }
         /** Constructs a set of options with the given type and strategy. */
-        scheduler_options_t(stream_type_t stream_type, schedule_strategy_t strategy)
-            : how_split(stream_type)
-            , strategy(strategy)
+        scheduler_options_t(mapping_t mapping, inter_input_dependency_t deps,
+                            int verbosity = 0)
+            : mapping(mapping)
+            , deps(deps)
+            , verbosity(verbosity)
         {
         }
 
         /** Size of the struct for binary-compatible additions. */
         size_t struct_size = sizeof(scheduler_options_t);
-        /** The base mode of the scheduler with respect to input streams. */
-        stream_type_t how_split = STREAM_BY_SYNTHETIC_CPU;
-        /** The scheduling strategy. */
-        schedule_strategy_t strategy = SCHEDULE_BY_INSTRUCTION_QUANTUM;
+        /** The mapping of inputs to outputs. */
+        mapping_t mapping = MAP_TO_RECORDED_OUTPUT;
```

Comment:
Since this is mapping has the strict requirement on core count matching the trace, would it make more sense to default to something flexible like MAP_TO_ANY_OUTPUT instead?